master/master.cfg

```diff
@@ -182,6 +182,46 @@ def is_important_change(change):
     return any(is_important_file(filename) for filename in change.files)
 
 
+def is_within_time_range(now, start, end):
+    if start <= end:
+        return start <= now <= end
+    else:
+        return now >= start or now <= end
+
+
+def get_delay(now, end):
+    today = datetime.today()
+    now = datetime.combine(today, now)
+    end = datetime.combine(today, end)
+
+    if now > end:
+        end += timedelta(days=1)
+
+    difference = end - now
+    return difference.total_seconds()
+
+
+# Avoid a build to be started between start and end time and delay such build
+# at end time
+def no_builds_between(start, end):
+    now = datetime.now().time()
+    start = datetime.strptime(start, "%H:%M").time()
+    end = datetime.strptime(end, "%H:%M").time()
+    def f(builder, requests):
+        if is_within_time_range(now, start, end):
+            delay = get_delay(now, end)
+            # Schedule the build later
+            builder.master.reactor.callLater(
+                int(delay),
+                builder.buildset_manager.submitBuildSet,
```

Comment:
We're getting an exception here:
```
    Traceback (most recent call last):
      File "/srv/buildbot/venv/lib/python3.9/site-packages/twisted/internet/defer.py", line 980, in _startRunCallbacks
        self._runCallbacks()
      File "/srv/buildbot/venv/lib/python3.9/site-packages/twisted/internet/defer.py", line 1074, in _runCallbacks
        current.result = callback(  # type: ignore[misc]
      File "/srv/buildbot/venv/lib/python3.9/site-packages/twisted/internet/defer.py", line 1960, in _gotResultInlineCallbacks
        _inlineCallbacks(r, gen, status, context)
      File "/srv/buildbot/venv/lib/python3.9/site-packages/twisted/internet/defer.py", line 2014, in _inlineCallbacks
        result = context.run(gen.send, result)
    --- <exception caught here> ---
      File "/srv/buildbot/venv/lib/python3.9/site-packages/buildbot/process/buildrequestdistributor.py", line 262, in _getNextUnclaimedBuildRequest
        nextBreq = yield self.nextBuild(self.bldr, breqs)
      File "/srv/buildbot/master/master.cfg", line 216, in f
        builder.buildset_manager.submitBuildSet,
    builtins.AttributeError: 'Builder' object has no attribute 'buildset_manager'
```

Reply:
If I'm reading the [nextBuild](https://docs.buildbot.net/4.1.0/manual/configuration/builders.html#:~:text=nextBuild) and Twisted docs correctly, this should return a Deferred like this:
```py
from twisted.internet import defer
...
            deferred = defer.Deferred()
            builder.master.reactor.callLater(
                int(delay),
                deferred.callback,
                requests[0],
            )
            return deferred
```
But, it would be nice to get confirmation from someone who's used Twisted before :)

Reply:
I filed #553